libsolidity/ast/ASTAnnotations.h

```diff
@@ -152,16 +152,21 @@ struct StructDeclarationAnnotation: TypeDeclarationAnnotation
 	std::optional<bool> containsNestedMapping;
 };
 
-struct ContractDefinitionAnnotation: TypeDeclarationAnnotation, StructurallyDocumentedAnnotation
+struct SharedContractDependenciesAnnotation
+{
+	/// List of contracts that this contract or function creates, i.e. contracts which need to be compiled first.
+	/// For ContractDefinitions: Also includes all contracts from linearizedBaseContracts.
+	/// Only relevant for contracts or free functions
+	std::map<ContractDefinition const*, ASTNode const*> contractDependencies;
```

Comment:
Can we be more specific than `ASTNode`? I think it can only be `NewExpression` or `InheritanceSpecifier`. Using a `variant` here might save us some assertions.